test/built-ins/Promise/prototype/finally/then-catch-finally-realm.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2020 Tooru Fujisawa. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: >
+  Promise.prototype.finally creates thenFinally and catchFinally in onFinally's
+  realm
+esid: sec-promise.prototype.finally
+info: |
+  Promise.prototype.finally ( _onFinally_ )
+  6. Else,
+    a. Let realm be ? GetFunctionRealm(_onFinally_).
+    c. Let _thenFinally_ be ! CreateBuiltinFunction(_stepsThenFinally_, &laquo; [[Constructor]], [[OnFinally]] &raquo;, realm).
+    g. Let _catchFinally_ be ! CreateBuiltinFunction(_stepsCatchFinally_, &laquo; [[Constructor]], [[OnFinally]] &raquo;, realm).
+  7. Return ? Invoke(_promise_, *"then"*, &laquo; _thenFinally_, _catchFinally_ &raquo;).
+features: [Promise.prototype.finally]
+---*/
+
+var other = $262.createRealm().global;
+
+var onFinally = () => {};
+
+var [thenFinally, catchFinally] = other.Promise.prototype.finally.call({
```

Comment:
Asserting the values returned by `Promise.prototype.finally` expands the scope of this test to include semantics which are unrelated to the normative change. That creates an opportunity (however unlikely) for implementations to spuriously pass if they modify the value returned by the `then` method. More importantly, it makes the test more challenging for human readers to understand.